public/js/home.js

```diff
@@ -52,6 +52,101 @@ window.addEventListener("click", function(event) {
+    let cssMode = (window.matchMedia('(prefers-color-scheme: dark)').matches);
+    let currentMode = "";
+    if(cssMode == true) {
+        currentMode = "dark";
+        // sets the slider to match the actual mode
+        slider.checked = true;
+        }
+    else {
+        currentMode = "light";
+    }
+    if (localStorage.getItem("color-scheme") === null) {
+        var prefferedMode = currentMode
+      }
+    else {
+        var prefferedMode = localStorage.getItem('color-scheme');
+    }
+    // sets mode to preffered if it's different than the current one
+    if (prefferedMode !== currentMode) {
+        slider.click();
+        currentMode = prefferedMode;
+    }
+    return currentMode;
+};
+
+// Toggle dark mode
+let dark = true;
+function darkMode() {
+    let r = document.querySelector(':root');
+    let rs = getComputedStyle(r);
+    // If the color variables are set to light mode, toggle dark mode
+    if (rs.getPropertyValue('--white') === "#ffffff" || rs.getPropertyValue('--white') === " #ffffff") {
+        r.style.setProperty('--green', '#00a336');
+        r.style.setProperty('--green1', '#00b83d');
+        r.style.setProperty('--green2', '#00cc44');
+        r.style.setProperty('--green3', '#00ff52');
+
+        r.style.setProperty('--white', '#141415');
+        r.style.setProperty('--white1', '#1e1f1f');
+        r.style.setProperty('--white2', '#28292a');
+        r.style.setProperty('--white3', '#323334');
+
+        r.style.setProperty('--gray', '#3c3d3e');
+        r.style.setProperty('--gray1', '#464749');
+        r.style.setProperty('--gray2', '#5a5c5e');
+        r.style.setProperty('--gray3', '#6e7072');
+        r.style.setProperty('--gray-to-white', "#cccccc");
+
+
+        r.style.setProperty('--black', '#ffffff');
+
+        r.style.setProperty('--blue1', '#1a3e75');
+        r.style.setProperty('--blue3', '#2c6bc9');
+
+        localStorage.setItem("color-scheme", "dark");
+        dark = true;
+    }
+    // Otherwise, toggle back to light mode
+    else {
+        r.style.setProperty('--green', '#268a48');
+        r.style.setProperty('--green1', '#107031');
+        r.style.setProperty('--green2', '#00551d');
+        r.style.setProperty('--green3', '#003913');
+
+        r.style.setProperty('--white', '#ffffff');
+        r.style.setProperty('--white1', '#f7f7f7');
+        r.style.setProperty('--white2', '#f2f2f2');
+        r.style.setProperty('--white3', '#dddddd');
+
+        r.style.setProperty('--gray', '#cccccc');
+        r.style.setProperty('--gray1', '#888888');
+        r.style.setProperty('--gray2', '#666666');
+        r.style.setProperty('--gray3', '#333333');
+        r.style.setProperty('--gray-to-white', "#666666");
+
+
+        r.style.setProperty('--black', '#000000');
+
+        r.style.setProperty('--blue1', '#9ebcea');
+        r.style.setProperty('--blue3', '#1d68cd');
+
+        localStorage.setItem("color-scheme", "light");
+        dark = false;
+    }
+ }
```

Comment:
This should be substantially reworked. The status quo is that the color definitions are repeated between `home.css` and `home.js`. Instead, `:root` and `@media (prefers-color-scheme: dark) { :root { ... } }` can be changed to be `body` and `body.dark`, and we can set/unset the `dark` class in JS according to `window.matchMedia('(prefers-color-scheme: dark)').matches` and the setting in `localStorage` (if any). Then, this function would simply need to toggle the `dark` class on the `body` element.